app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php

```diff
@@ -51,62 +52,50 @@ public function getAllCountries()
         return $activeCountries;
     }
 
-    public function projectsCountry($country)
+    public function numberOfProjects($country, $projects)
     {
-        $query = Project::where('framework_key', 'terrafund')
-            ->whereHas('organisation', function ($query) {
-            $query->whereIn('type', ['for-profit-organization', 'non-profit-organization']);
-        });
-        return $query->where('country', $country);
+        return $projects->where('country', $country)->count();
     }
 
-    public function numberOfProjects($country)
+    public function totalSpeciesAmount($country, $projects)
     {
-        return $this->projectsCountry($country)->count();
-    }
+        $projects = $projects->where('country', $country);
 
-    public function totalSpeciesAmount($country)
-    {
-        $projects = $this->projectsCountry($country)->get();
-        $totalSpeciesAmount = 0;
-        foreach ($projects as $project) {
-            $sites = $project->sites()->pluck('id')->toArray();
-            foreach ($sites as $siteId) {
-                $latestSiteReportId = SiteReport::where('site_id', $siteId)
-                    ->orderByDesc('due_at')
-                    ->value('id');
-                if ($latestSiteReportId !== null) {
-                    $totalSpeciesAmount += TreeSpecies::where('speciesable_id', $latestSiteReportId)->sum('amount');
+        return $projects->sum(function ($project) {
+            return $project->sites()->get()->sum(function ($site) {
+                $latestReport = $site->reports()->orderByDesc('due_at')->first();
+                if ($latestReport) {
+                    return $latestReport->treeSpecies()->sum('amount');
                 }
```

Comment:
Since this method is being implicitly typed to return an integer, this should likely `return 0;` after the if block for clarity.